src/angularComponent/ngcOmniboxController.js

```diff
@@ -104,7 +92,19 @@ export default class NgcOmniboxController {
     this._ngModel = newModel;
 
     if (Array.isArray(this._ngModel)) {
-      Object.setPrototypeOf(this._ngModel, customArrayPrototype);
+      const currentPrototype = Object.getPrototypeOf(this._ngModel);
+      const newPrototype = Object.create(currentPrototype);
+      Object.setPrototypeOf(this._ngModel, newPrototype);
```

Comment:
Maybe this line should be done after you override those methods instead of before? Not sure if it matters, but somehow it feels more safe to me.